src/mlpack/core/cv/meta_info_extractor.hpp

```diff
@@ -0,0 +1,249 @@
+/**
+ * @file meta_info_extractor.hpp
+ * @author Kirill Mishchenko
+ *
+ * Tools for extracting meta information about machine learning algorithms.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_META_INFO_EXTRACTOR_HPP
+#define MLPACK_CORE_CV_META_INFO_EXTRACTOR_HPP
+
+#include <type_traits>
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/sfinae_utility.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/*
+ * A wrapper struct for holding a Train form.
+ *
+ * @tparam MatType The type of data.
+ * @tparam PredictionsType The type of predictions.
```

Comment:
The assumption here is that `LabelsType == PredictionsType`, right?  I think that's completely reasonable, I just want to make sure I am comprehending fully.